llvm/lib/Transforms/Instrumentation/LowerAllowCheckPass.cpp

```diff
@@ -81,19 +83,32 @@ static bool removeUbsanTraps(Function &F, const BlockFrequencyInfo &BFI,
     return *Rng;
   };
 
-  auto ShouldRemoveHot = [&](const BasicBlock &BB) {
-    return HotPercentileCutoff.getNumOccurrences() && PSI &&
-           PSI->isHotCountNthPercentile(
-               HotPercentileCutoff, BFI.getBlockProfileCount(&BB).value_or(0));
+  auto GetCutoff = [&](const IntrinsicInst *II) {
+    unsigned int cutoff = 0;
+    if (HotPercentileCutoff.getNumOccurrences())
+      cutoff = HotPercentileCutoff;
```

Comment:
Return ASAP is easy to read
`return HotPercentileCutoff;`

Reply:
Done